span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/SpanFilter.java

```diff
@@ -147,4 +183,32 @@ private boolean isRootExitSpan(
 
     return SPAN_KIND_CLIENT.equals(spanKindKeyValue.getVStr());
   }
+
+  private boolean anySpanDropFiltersMatch(
+      List<List<SpanDropFilter>> spanDropFilters,
+      Map<String, JaegerSpanInternalModel.KeyValue> tags) {
+    return spanDropFilters.stream()
+        .anyMatch(
+            andFilters ->
+                andFilters.stream().allMatch(filter -> matchSpanDropFilter(filter, tags)));
+  }
+
+  private boolean matchSpanDropFilter(
+      SpanDropFilter filter, Map<String, JaegerSpanInternalModel.KeyValue> tags) {
+    switch (filter.getOperator()) {
+      case EQ:
+        return tags.containsKey(filter.getTagKey())
+            && StringUtils.equals(tags.get(filter.getTagKey()).getVStr(), filter.getTagValue());
```

Comment:
@kotharironak We may have to look with the process tags too. See below

```
  public Optional<String> getTenantId(
      Map<String, KeyValue> spanTags, Map<String, KeyValue> processTags) {
    return Optional.ofNullable(processTags.get(tenantIdKey))
        .or(() -> Optional.ofNullable(spanTags.get(tenantIdKey)))
        .map(KeyValue::getVStr);
  }
```

Reply:
So, far we are not checking process tags, and we are able to filter, right?

Reply:
@ravisingal @avinashkolluru I have added the support for processTags too. So, now the filter will check in the union of two tags (`processTags`,` spanTags`).

Total Tags = `Process Tags` + `Span Tags` will be used for matching a filter.